android/intltestapp/cpp/ReplProxy.cpp

```diff
@@ -0,0 +1,195 @@
+        method(javaClassStatic(), facebook::jni::make_jstring(info));
+    }
+
+    ReplProxy() {
+        vm::RuntimeConfig config = vm::RuntimeConfig::Builder()
+                .withGCConfig(
+                        vm::GCConfig::Builder()
+                                .withInitHeapSize(32 << 20)
+                                .withMaxHeapSize(512 << 20)
+                                .withSanitizeConfig(vm::GCSanitizeConfig::Builder()
+                                                            .withSanitizeRate(0.0)
+                                                            .withRandomSeed(-1)
+                                                            .build())
+                                .withShouldRecordStats(false)
+                                .withCallback([this](vm::GCEventKind gcEventKind,
+                                                     const char *extraInfo) {
+                                    GCCallback(gcEventKind, extraInfo);
+                                })
+                                .build())
+                .withVMExperimentFlags(vm::RuntimeConfig::getDefaultVMExperimentFlags())
+                .withES6Promise(vm::RuntimeConfig::getDefaultES6Promise())
+                .withES6Proxy(vm::RuntimeConfig::getDefaultES6Proxy())
+                .withES6Symbol(vm::RuntimeConfig::getDefaultES6Symbol())
+                // .withES6Intl(true)
+                .withEnableHermesInternal(true)
+                .withEnableHermesInternalTestMethods(true)
+                .withAllowFunctionToStringWithRuntimeSource(false)
+                .build();
+
+        mRuntime = vm::Runtime::create(config);
+
+    }
+
+    static jni::local_ref<ReplProxy::jhybriddata> initHybrid(jni::alias_ref<jclass>) {
+        return makeCxxInstance();
+    }
+
+    void nativeCollect(facebook::jni::alias_ref<facebook::jni::JString> jCause) {
+        mRuntime->collect(jCause->toStdString());
+    }
+
+    std::string nativeHeapStats() {
+        std::string s;
+        llvh::raw_string_ostream os(s);
+        mRuntime->printHeapStats(os);
+        return s;
+    }
+
+    std::string nativeEvalScript(facebook::jni::alias_ref<facebook::jni::JString> jScript) {
+        std::string script = jScript->toStdString();
+        std::string response;
+        response.reserve(1024);
+
+        vm::GCScope gcScope(mRuntime.get());
+
+        vm::Handle <vm::JSObject> global = mRuntime->getGlobal();
+        auto propRes = vm::JSObject::getNamed_RJS(
+                global, mRuntime.get(), vm::Predefined::getSymbolID(vm::Predefined::eval));
+        if (propRes == vm::ExecutionStatus::EXCEPTION) {
+            mRuntime->printException(
+                    llvh::outs(), mRuntime->makeHandle(mRuntime->getThrownValue()));
+            return "error getting 'eval' from global";
+        }
+        auto evalFn = mRuntime->makeHandle<vm::Callable>(std::move(*propRes));
```

Comment:
Is it possible to cache `evalFn` when we create the Runtime so we don't have to look it up each time?

Reply:
Done in the latest iteration ..